src/index/import/markdown.ts

```diff
@@ -381,3 +410,43 @@ export function parseFrontmatter(value: any): Literal {
     // Backup if we don't understand the type.
     return null;
 }
+/** recursively parse yaml links and return them as a single array */
+export function parseYAMLLinks(obj: any): Link[] {
+    let links: Link[] = [];
+    for (let v in obj) {
+        let obj2 = parseFrontmatter(obj[v]);
+
+        if (typeof obj2 === "object") {
+            if (Array.isArray(obj2)) {
+                let result = [];
+                for (let child of obj2 as Array<any>) {
+                    result.push(parseFrontmatter(child));
+                }
```

Comment:
You don't need to make this intermediate `result` - just iterate over `obj2` directly in the for loop below and call `parseFrontmatter` in the loop.